src/js/containers/agencyV2/AgencyContainerV2.jsx

```diff
@@ -15,28 +15,33 @@ import { setAgencyOverview } from 'redux/actions/agencyV2/agencyV2Actions';
 
 import { useValidTimeBasedQueryParams, useLatestAccountData } from 'containers/account/WithLatestFy';
 import AgencyPage from 'components/agencyV2/AgencyPage';
+import { useAgencySlugs } from './WithAgencySlugs';
 
 export const AgencyProfileV2 = () => {
-    const { agencyId } = useParams();
+    const { agencySlug } = useParams();
     const [, , { year: latestFy }] = useLatestAccountData();
     const { fy: currentUrlFy } = useQueryParams(['fy']);
     const [selectedFy, setSelectedFy] = useValidTimeBasedQueryParams(currentUrlFy, null, ['fy']);
+    // Use a custom hook to get the { agency slug: toptier code } mapping, or request it if not yet available
+    const [agencySlugs, slugsLoading, slugsError] = useAgencySlugs();
     const [isLoading, setLoading] = useState(true);
     const [isError, setError] = useState(false);
     const [errorMessage, setErrorMessage] = useState('');
+    const [redirect, setRedirect] = useState(false);
     const request = useRef(null);
     const dispatch = useDispatch();
+    const [toptierCode, setToptierCode] = useState(agencySlugs[agencySlug]);
 
     useEffect(() => {
-        if (selectedFy) {
+        if (selectedFy && toptierCode) {
```

Comment:
Question for my own knowledge! why do we now need to include toptierCode in this useEffect when previously we just needed selectedFY to make the request?

Reply:
Good question! Before, we were getting toptier code (agency ID) from the URL parameter. So as soon as you landed on the page, it was available in the URL. Now, we have to wait for the API request made by `useAgencySlugs` to complete before the toptierCode is available.